config/hsc/jointcal.py

```diff
@@ -7,3 +7,16 @@
 config.applyColorTerms = True
 
 config.colorterms.load(os.path.join(getPackageDir("obs_subaru"), "config", "hsc", "colorterms.py"))
+
+config.sourceSelector.name = 'science'
+config.sourceSelector['science'].doUnresolved = True
+config.sourceSelector['science'].doSignalToNoise = True
+config.sourceSelector['science'].signalToNoise.minimum = 10.
+config.sourceSelector['science'].doIsolated = False
+config.sourceSelector['science'].doFlags = True
+config.sourceSelector['science'].doFluxLimit = True
+config.sourceSelector['science'].fluxLimit.minimum = 10
+badFlags = ['base_PixelFlags_flag_edge', 'base_PixelFlags_flag_saturated',
+            'base_PixelFlags_flag_interpolatedCenter', 'base_SdssCentroid_flag',
+            'base_PsfFlux_flag', 'base_PixelFlags_flag_suspectCenter']
```

Comment:
Jointcal also uses `slot_Shape_flag` in its badFlags list. I don't know whether that is redundant with the others or not.

Reply:
That's true only if the user is using the `astrometry` source selector`!  I am not!  The name difference should make this more obvious.  (`flags.bad` not `badFlags`)

Your `JointcalConfig.setDefaults` is highly nonstandard btw.  We the select default source selectors by doing `self.sourceSelector.name = astrometry`. `astrometry` is now "active."  If your desired config change applies to all source selectors in the registry, you can access the "active" source selector's config via `self.sourceSelector.active.`  If that's not guaranteed, I specify which one it applies to `self.sourceSelector['astrometry'].badFlags.extend(["slot_Shape_flag"])`.  For example if I were to do `self.sourceSelector.active.badFlags.extend(["slot_Shape_flag"])` in the file here,  it'd fail because `science` doesn't have a `badFlags`

Reply:
What do you mean by "highly nonstandard" in this context? Can you show me the documentation on what a `config.setDefaults` is supposed to look like?

My comment above was just that you have a bad flags list here that did not include `slot_Shape_flag`, and I was wondering if it should be in your list or no.t

Reply:
Sure. Your use of `setDefaults` is fine, it is the setting of the  *RegistryField* that is weird:

https://pipelines.lsst.io/modules/lsst.pipe.base/command-line-task-retargeting-howto.html#how-to-retarget-a-subtask-configured-as-a-registryfield-with-a-configuration-file

Reply:
Sorry, I did misunderstand your comment about missing the `slot_Shape_flag`.  I assumed it was necessary if you were including galaxies, but not if you were selecting bright point sources.